lib/create-app.js

```diff
@@ -107,66 +82,88 @@ function createApp (argv = {}) {
     // adds POST /api/accounts/recover
     // adds GET /api/accounts/validateToken
     app.use('/api/accounts/', accountRecovery)
-  }
 
-  // Adding Multi-user support
-  if (ldp.webid) {
-    var idp = IdentityProvider({
+    let accountManager = AccountManager.from({
+      authMethod: argv.auth,
+      emailService: app.locals.email,
+      host: argv.host,
+      accountTemplatePath: argv.accountTemplatePath,
       store: ldp,
-      suffixAcl: ldp.suffixAcl,
-      suffixMeta: ldp.suffixMeta,
-      settings: 'settings',
-      inbox: 'inbox',
-      auth: ldp.auth
+      multiUser
     })
 
-    var needsOverwrite = function (req, res, next) {
-      checkMasterAcl(req, function (found) {
-        if (!found) {
-          // this allows IdentityProvider to overwrite root acls
-          idp.middleware(true)(req, res, next)
-        } else if (found && ldp.idp) {
-          idp.middleware(false)(req, res, next)
-        } else {
-          next()
-        }
-      })
-    }
+    // Account Management API (create account, new cert)
+    app.use('/', API.userAccounts.middleware(accountManager))
 
-    // adds POST /api/accounts/new
-    // adds POST /api/accounts/newCert
-    app.get('/', idp.get.bind(idp))
-    app.use('/api/accounts', needsOverwrite)
-    app.post('/api/accounts/signin', bodyParser.urlencoded({ extended: false }), API.accounts.signin())
-    app.post('/api/accounts/signout', API.accounts.signout())
-    app.post('/api/messages', authentication, bodyParser.urlencoded({ extended: false }), API.messages.send())
+    // Authentication API (login/logout)
+    app.post('/api/accounts/signin', bodyParser, API.authn.signin())
+    app.post('/api/accounts/signout', API.authn.signout())
+
+    // Messaging API
+    app.post('/api/messages', authentication, bodyParser, API.messages.send())
```

Comment:
I see. This would also remove the need for that API index module to re-export all those other api modules.